napi-inl.h

```diff
@@ -731,55 +799,74 @@ inline Number::Number(napi_env env, napi_value value) : Value(env, value) {
 }
 
 inline Number::operator int32_t() const {
+#ifdef NODE_ADDON_API_ENABLE_MAYBE
+  return Int32Value().FromMaybe(0);
```

Comment:
Like said, as the function is designed as a convenience method, a default value is returned if the box is empty.